src/tool/cpp/cppwinrt/strings/base_implements.h

```diff
@@ -1284,6 +1291,44 @@ namespace winrt
         }
 
     private:
+
+#ifdef _DEBUG
+        bool is_stack_address(void const* const address) noexcept
+        {
+            struct memory_basic_information
+            {
+                void* BaseAddress;
+                void* AllocationBase;
+                uint32_t AllocationProtect;
+                size_t RegionSize;
+                uint32_t State;
+                uint32_t Protect;
+                uint32_t Type;
+            };
+
+            static constexpr uint32_t page_guard{ 0x100 };
+
+            memory_basic_information info{};
+            WINRT_VirtualQuery(address, &info, sizeof(info));
+
+            if (info.AllocationBase == info.BaseAddress)
+            {
+                return (info.Protect & page_guard) == page_guard;
+            }
+
+            WINRT_VirtualQuery(info.AllocationBase, &info, sizeof(info));
+
+            if ((info.Protect& page_guard) == page_guard)
+            {
+                return true;
+            }
+
+            void const* const potential_page = static_cast<char const*>(info.AllocationBase) + info.RegionSize;
```

Comment:
`potential_guard_page`?  (It's definitely a page--no potentiality about that :-))

Reply:
A bit too much brevity there on my part. 😉